client/javascript/src/httpclient.js

```diff
@@ -0,0 +1,79 @@
+export default class HttpClient {
+    constructor(options = {}) {
+      this._baseURL = options.baseURL || "";
+      this._headers = options.headers || {};
+    }
+  
+    async _fetchJSON(endpoint, options = {}) {
+      const res = await fetch(this._baseURL + endpoint, {
+        ...options,
+        headers: this._headers
+      });
+  
+      if (!res.ok) throw new Error(res.statusText);
+  
+      if (options.parseResponse !== false && res.status !== 204)
+        return res.json();
+  
+      return undefined;
+    }
+  
+    setHeader(key, value) {
+      this._headers[key] = value;
+      return this;
+    }
+  
+    getHeader(key) {
+      return this._headers[key];
+    }
+  
+    setBasicAuth(username, password) {
+      this._headers.Authorization = `Basic ${btoa(`${username}:${password}`)}`;
+      return this;
+    }
+  
+    setBearerAuth(token) {
+      this._headers.Authorization = `Bearer ${token}`;
+      return this;
+    }
+  
+    get(endpoint, options = {}) {
```

Comment:
Should we just have all of these be async as well? Then it's up to the user of the client to use `await` or indeed run it async if they want to.

Reply:
Good idea, updated!